connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala

```diff
@@ -265,6 +271,12 @@ case class SessionHolder(userId: String, sessionId: String, session: SparkSessio
 
 object SessionHolder {
 
+  // The maxmimum number of errors in the cache.
+  private val ERROR_CACHE_SIZE = 20
+
+  // The maximum time for an error to stay in the cache.
+  private val ERROR_CACHE_TIMEOUT_SEC = 60
+
```

Comment:
is this 20 distinct exceptions or 20 stacks?

Reply:
20 distinct exceptions